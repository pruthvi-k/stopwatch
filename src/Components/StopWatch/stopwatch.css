@media only screen and (max-width: 600px) {
  .stopwatch {
    width: 100%;
  }
  .stopwatch .timer {
    width: 52%;
  }
}

@media only screen and (min-width: 599px) {
  .stopwatch > div {
    width: 20% !important;
    /* font-size: 20px; */
  }
}
.stopwatch {
  font-size: 1.5em;
  background-color: rgba(248, 247, 216, 0.7) !important;
}
.stopwatch .timer {
  padding-top: 20%;
  border: 6px solid rgba(35, 39, 43, 0.9);
  height: 20%;
  position: relative;
  margin-bottom: 10px;
  font-weight: bold;
  background-color: rgba(255, 255, 255, 0.3);
}
.timer-text {
  position: absolute;
  top: 42%;
  width: 100%;
}
