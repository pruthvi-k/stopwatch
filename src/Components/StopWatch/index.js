import React, { Component } from "react";
import { Button, Row, Col } from "react-bootstrap";
import "./stopwatch.css";

class StopWatch extends Component {
  state = {
    timerOn: false,
    start: 0,
    time: 0,
  };

  startTimer = () => {
    this.setState({
      timerOn: true,
      time: this.state.time,
      start: Date.now() - this.state.time,
    });
    this.timer = setInterval(() => {
      this.setState({
        time: Date.now() - this.state.start,
      });
    }, 10);
  };

  stopTimer = () => {
    this.setState({ timerOn: false });
    clearInterval(this.timer);
  };

  resetTimer = () => {
    this.setState({
      start: 0,
      time: 0,
    });
  };

  render() {
    const { time } = this.state;
    let seconds = ("0" + (Math.floor(time / 1000) % 60)).slice(-2);
    let minutes = ("0" + (Math.floor(time / 60000) % 60)).slice(-2);
    let hours = ("0" + Math.floor(time / 3600000)).slice(-2);
    return (
      <>
        {/* <div className="stopwatch-header">Stopwatch</div> */}

        <div className="stopwatch text-center d-flex align-items-center flex-column justify-content-center h-100 w-100 bg-dark">
          <div className="timer rounded-circle">
            <div className="timer-text">
              {hours} : {minutes} : {seconds}
            </div>
          </div>

          {this.state.timerOn === false && this.state.time === 0 && (
            <div>
              <Button
                onClick={this.startTimer}
                id="startButton"
                block
                className="my-1 btn-dark"
              >
                Start
              </Button>
            </div>
          )}

          {this.state.timerOn === true && (
            <div>
              <Button onClick={this.stopTimer} block className="my-1 btn-dark">
                Stop
              </Button>
            </div>
          )}
          {this.state.timerOn === false && this.state.time > 0 && (
            <div>
              <Button onClick={this.startTimer} block className="my-1 btn-dark">
                Resume
              </Button>
            </div>
          )}
          {this.state.timerOn === false && this.state.time > 0 && (
            <div>
              <Button onClick={this.resetTimer} block className="my-1 btn-dark">
                Reset
              </Button>
            </div>
          )}
        </div>
      </>
    );
  }
}

export default StopWatch;
