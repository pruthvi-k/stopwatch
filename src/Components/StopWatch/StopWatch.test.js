import React from "react";
import { shallow } from "enzyme";
import StopWatch from "./index";

describe("Stop Watch component", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<StopWatch />);
  });
  it("should render stopwatch component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("has the initial state count of zero", () => {
    expect(wrapper.state().timerOn).toEqual(false);
  });

  //   describe('The Count Up Button', () => {
  //     it('increments state count by 1 on click', () => {
  //       wrapper.find('#count-up').props().onClick();
  //       expect(wrapper.state()).toEqual({ count: 1 });
  //     });
  //   });

  //   describe('The Count Down Button', () => {
  //     it('decrements state count by 1 on click', () => {
  //       wrapper.find('#count-down').props().onClick();
  //       expect(wrapper.state()).toEqual({ count: -1 });
  //     });
  //   });
  it("should have start button", () => {
    wrapper.find("#startButton").simulate("click");
    expect(wrapper.state().timerOn).toEqual(true);
  });
});
