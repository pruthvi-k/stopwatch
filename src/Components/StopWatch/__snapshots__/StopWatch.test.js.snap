// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stop Watch component should render stopwatch component 1`] = `ShallowWrapper {}`;

exports[`should render stopwatch component 1`] = `ShallowWrapper {}`;
